api/src/schemas.py

```diff
@@ -185,12 +186,11 @@ def exists(cls, collection):
 
 
 class S3Input(WorkflowInputBase):
-    # s3 discovery
-    discovery: Literal[Discovery.s3]
-
+    discovery: Literal["s3"]
     prefix: str
     bucket: str
-    filename_regex: Optional[str]
+    filename_regex: str
```

Comment:
in the absence of `filename_regex`, the default behavior is to discover all the files in <bucket>/<prefix> which is a sensible default

is there a reason for making it required?

Reply:
Wasn't aware of this behaviour, I can adjust.